src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1482,14 +1490,6 @@ MulticopterPositionControl::control_non_manual(float dt)
 		control_auto(dt);
 	}
 
-	/* weather-vane mode for vtol: disable yaw control */
-	if (_vehicle_status.is_vtol) {
-		_att_sp.disable_mc_yaw_control = _pos_sp_triplet.current.disable_mc_yaw_control;
-
-	} else {
-		_att_sp.disable_mc_yaw_control = false;
-	}
-
```

Comment:
The VTOL controller intercepts the attitude setpoint publications. Now with this PR it will set that directly. Still not the overall architecture I want, but it's concentrating more of the relevant pieces.